test/integration/targets/docker_image/tasks/tests/options.yml

```
@@ -290,3 +290,28 @@
     that:
       - path_1 is changed
       - path_2 is not changed
+
+####################################################################
+## target ##########################################################
+####################################################################
+
+- name: Build multi-stage image
+  docker_image:
+    name: "{{ iname }}"
+    build:
+      path: "{{ role_path }}/files/StagedDockerfile"
+      target: first
+      pull: no
+    source: build
+  register: dockerfile_2
+
+- name: cleanup
+  docker_image:
+    name: "{{ iname }}"
+    state: absent
+    force_absent: yes
+
+- assert:
+    that:
+    - dockerfile_2 is changed
+    - dockerfile_2['image']['Config']['WorkingDir'] == '/first'
```

You can simplify this to
```suggestion
    - dockerfile_2.image.Config.WorkingDir == '/first'
```